rerun_py/rerun_sdk/rerun/__init__.py

```diff
@@ -226,6 +227,7 @@ def _init_recording_stream() -> None:
             notebook_show,
         ]
         + [
+            set_index,
             set_time_sequence,
             set_time_seconds,
             set_time_nanos,
```

Comment:
This makes me think that it might be worth opening a reminder "remove deprecated API" issue scheduled for 0.24 or 0.25, and ideally tag these places (though a global search will probably good enough when we actually do it).